src/GlobalStateManager.ts

```diff
@@ -1,13 +1,26 @@
-import type * as vscode from 'vscode';
+import * as vscode from 'vscode';
+
 export class GlobalStateManager {
     constructor(
         private context: vscode.ExtensionContext
-    ) { }
+    ) {
+        this.updateFromVsCodeConfiguration();
+        vscode.workspace.onDidChangeConfiguration(this.updateFromVsCodeConfiguration);
```

Comment:
This should work right? I've noticed in a bunch of spots we actually duplicate it all it seems to me most of the time we could have one touch point for reading the first time and when changes happen.